corehq/apps/hqadmin/views/data.py

```diff
@@ -39,32 +69,62 @@ def get(self, record_id):
             raise ResourceNotFound("missing")
 
 
-_SQL_DBS = OrderedDict((db.dbname, db) for db in [
-    _Db(
-        XFormInstanceSQL._meta.db_table,
-        lambda id_: XFormInstanceSQLRawDocSerializer(XFormInstanceSQL.get_obj_by_id(id_)).data,
-        XFormInstanceSQL.__name__
-    ),
-    _Db(
-        CommCareCaseSQL._meta.db_table,
-        lambda id_: CommCareCaseSQLRawDocSerializer(CommCareCaseSQL.get_obj_by_id(id_)).data,
-        CommCareCaseSQL.__name__
-    ),
-    _Db(
-        SQLLocation._meta.db_table,
-        lambda id_: SQLLocation.objects.get(location_id=id_).to_json(),
-        SQLLocation.__name__
-    ),
-])
+class _SQLFormDb(_SQLDb):
+    def get_context(self, record_id):
+        form = self.get(record_id)
+        doc = XFormInstanceSQLRawDocSerializer(form).data
+        context = self._context_for_doc(doc)
+        if 'form' not in doc:
+            context['errors'].append(_('Missing Form XML'))
+        elif not doc['form']:
```

Comment:
would invalid xml lead to blank value for this?

Reply:
yes, it would be an empty dict: https://github.com/dimagi/commcare-hq/blob/master/corehq/form_processor/models.py#L485